File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    answer = 0
    mine = []
    di = [-1, 0, 1]
    dj = [-1, 0, 1]
    for i in range(len(board)) :
        for j in range(len(board[i])) :
            if board[i][j] == 1 :
                mine.append([i, j])
    
    for i, j in mine :
        for d_i in di :
            for d_j in dj :
                ni = i + d_i
                nj = j + d_j
                
                if ni >= 0 and ni < len(board) and nj >= 0 and nj < len(board) :
                    board[ni][nj] = 1
            
    for i in board :
        print(i)
        answer += i.count(0)
    
    return answer

File: 프로그래머스/0/120866. 안전지대/안전지대.java
import java.util.*;

class Solution {
    private static final int[] dx = {-1, -1, -1, 0, 1, 1, 1, 0};
    private static final int[] dy = {-1, 0, 1, 1, 1, 0, -1, -1};
    
    public int solution(int[][] board) {
        int[][] visited = new int[board.length][board.length];

        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[i][j] == 1 && visited[i][j] == 0) {
                    bfs(board, i, j, visited);
                }
            }
        }
        
        int answer = count(board);
        return answer;
    }
    
    public static int count(int[][] map) {
        int answer = 0;
        for (int i = 0; i < map.length; i++) {
            for (int j = 0; j < map.length; j++) {
                if (map[i][j] == 0) {
                    answer++;
                }
            }
        }
        return answer;
    }

    public static void bfs(int[][] map, int i, int j, int[][] visited) {
        Queue<Integer[]> queue = new ArrayDeque<>();
        queue.add(new Integer[]{i, j});
        int width = map.length;
        int height = width;
        while (!queue.isEmpty()) {
            Integer[] remove = queue.remove();
            int x = remove[0];
            int y = remove[1];

            for (int k = 0; k < 8; k++) {
                int nx = x + dx[k];
                int ny = y + dy[k];

                if (0 <= nx && nx < width && 0 <= ny && ny < height) {
                    if (map[nx][ny] == 0 && visited[nx][ny] == 0) {
                        map[nx][ny] += 1;
                        visited[nx][ny] = 1;
                    }
                }
            }
        }
    }
}